### DISCUSSION TITLE
mouse moved to modal dialog on another screen

### DISCUSSION THREAD

**Discussion by shiroyagi** [i3/i3#6104] [Q&A] [answered]
If an application is showing a modal dialog then it seems i3 moves the mouse pointer to that dialog when the application gains focus.

If there are two such applications side by side each with their modal dialog above the other application, it becomes very hard to reach particular dialogs as the mouse jumps away in a confusing way.  The only way to recover seems to be to open a terminal and kill one of them.

Is it possible to disable this behaviour so the mouse is not moved in this way?  Other WMs don't seem to do this.

**Comment by okraits** [comment]
Are you sure that i3 does that? I didn't experience something like that so far. Can you give examples of applications where this happens for you?

**Reply by shiroyagi** [reply]
I don't know what else could be responsible for moving the mouse, it is not the application and it does not happen in Plasma(KDE).

It happens with an application using QMessageBox::exec() that was constructed with a QMainWindow as its parent.

eg.
- Open Qt5 Assistant, Help->About
- When the About Dialog is on the same screen as the main application, the mouse is not grabbed
- Move the About Dialog onto another screen, then the mouse over the main application
- The mouse jumps back to the About Dialog

**Reply by orestisfl** [reply, marked as answer]
i3 only touches the cursor if `mouse_warping` is used, see https://i3wm.org/docs/userguide.html#_mouse_warping. Try
```
mouse_warping none
```
and see if the issue persists.

**Reply by orestisfl** [reply]
From a quick experiment with Qt Assistant and your instructions (thank you for being detailed), the `mouse_warping` setting seems to be causing the mouse movement, since the window is focused on a different workspace.

**Reply by shiroyagi** [reply]
Turning off `mouse_warping` does avoid this issue, but in many situations I would prefer it on.  Possibly when focus is changed by the keyboard I would like warping and when by moving the mouse I would not.  Maybe a new feature request?
